drivers/include/can/candev.h

```diff
@@ -161,6 +161,19 @@ typedef struct candev_driver {
      */
     int (*set_filter)(candev_t *dev, const struct can_filter *filter);
 
+#if defined(MODULE_CAN_FILTER_TYPE_SPECIFIC)
+    /**
+     * @brief   Set and configure a reception filter
+     *
+     * @param[in] dev           CAN device descriptor
+     * @param[in] filter_mode   The mode in which the filter will be applied
+     *
+     * @return              a positive filter number
+     * @return              <0 on error
+     */
+    int (*set_filter_type_spec)(candev_t *dev, const struct can_filter_mode *filter_mode);
```

Comment:
We will probably need to do some changes for other CPUs (STM for example) then. The reason I left it is to provide the user the opportunity to use the `.set_filter` in case of usage of CAN controller not providing different CAN filters configuration., and the new one as a sort of an extension